segments-guide.md

```diff
@@ -0,0 +1,327 @@
+# Segments APIs
+
+## Segments Overview
+
+The Analytics 2.0 Segments APIs allow you to retrieve, update, or create segments programmatically through Adobe I/O. The APIs use the same data and methods that are used when working with segments in the UI. For more information on Adobe Analytics segments, see [Adobe Help](https://marketing.adobe.com/resources/help/en_US/analytics/segment/seg_overview.html). For more information on understanding segment definition data architecture, see the [Segment Definition](segments.md) article that accompanies this user guide.
+
+## Authorization and Authentication
+
+To obtain authorization and authentication, see the [Getting Started Guide for 2.0 APIs](https://www.adobe.io/apis/experiencecloud/analytics/docs.html#!AdobeDocs/analytics-2.0-apis/master/create-oauth-client.md).
+
+## Best Practices
+
+Please follow these guidelines when using the Segments APIs:
+
+*  Make multiple, smaller requests instead of a large, single request.
+*  Request data once and cache it.
+*  Use caution when updating a segment so that you do not alter the original segment used by others.
+*  Avoid creating duplicate segments with the same definition. Creating many segments will affect performance for your company in some situations.
+*  Use the Analytics UI for creating segments from scratch, if possible. Because of the flexibility in using segments, the UI offers significant advantages for managing and optimizing the complexity of this task. Consider creating a segment in the UI that matches, as closely as possible, the segment you want, then modifying the JSON so that it does what you want programmatically. Then you can `POST` the JSON body of that modified segment to the API.
+
+## /segments Endpoint Description
+
+The `/segments` endpoint description is shown in our [Swagger UI](https://adobedocs.github.io/analytics-2.0-apis/). Use the Swagger UI to see endpoint summaries, available methods, parameters, example values, models, and status codes, and to try out the API.
+
+## Templates
+
+Adobe provides many predefined segments, or templates, for clients in specific situations. Although original templates cannot be modified, copies of them can be made and then the copies can be modified. If you want to work with segments templates programmatically, you can identify them by the additional attribute `template: true` as shown at the bottom of the following example:
```

Comment:
Do we let people create segment templates in the API? The last I remember this was not possible.

Reply:
Kent used the following language.

> There are many predefined segments that Adobe provides to its clients for use. These are segments that have been useful to many clients in specific situations. These segments can not be modified in any way, or deleted. These segments can differentiated because they have an additional attribute like “template”:true.


My reading of your sentence made me feel like I could programmatically create a segment template. You specifically stated that I couldn't modify existing ones, but it seemed like I could programatically use that template:true flag.

> If you want to work with segments templates programmatically, you can identify them by the additional attribute `template: true` as shown at the bottom of the following example:

It may be more clear if you shortened the sentence
> You can identify that as segment is a template by the additional attribute `template: true` as shown at the bottom of the following example:

Reply:
Thanks, I will shorten the sentence. But I would like to make sure I understand the language suggestion.

Is it the wording "working with segments templates programmatically" that makes it sound like you can "create" the segments template? I thought "working with segments templates" programmatically is not the same thing as "creating" segments.  If you change the json a bit to modify a template for your needs, aren't you doing this so that you can "work with segments templates programmatically?" 

Here is my correspondence with Kent:

John Wight [6:09 PM]
Hi Kent,  In the "Templates" section, you state that they "cannot be modified in any way." I tried to learn more about these templates and found in help the following sentence: "You can drag these templates into the Definitions canvas and use them as they have been defined, or modify them." Does this contradict the template features or am I just reading about two different things? Thanks for any help. John Wight.

John Wight [6:18 PM]
Also, regarding the "template: true" attribute, do you know where I could find a sample that shows that. It might be helpful for me to include it with the guide. Thanks.

Kent Otis [8:46 AM]
When a template is modified, a copy is mode, then modified.

Reply:
It is actually the "you can identify them" portion that made me wonder how I can use this. It made we feel like I could programmatically use this identifier in the APIs(possibly for a creation). 

A lot of people read the docs and there is a lot of opportunity for misinterpretation. I figure that if I got that drift scanning through that others would as well and it is better to be very clear rather than having unhappy readers, questions, and support tickets.